spec.bs

```diff
@@ -3485,6 +3492,57 @@ A <dfn>server auction response</dfn> is a [=struct=] that contains auction resul
   :: A [=map=] whose [=map/keys=] are [=origins=] and whose [=map/values=] are [=lists=] of [=urls=].
 </dl>
 
+<dfn>server auction join candidates</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="server auction join candidates">
+  : <dfn>ad render url hash</dfn>
+  :: A [=SHA-256=] hash of the [=k-anonymity key=] for the ad in the winning bid,
+     as calculated using [=compute the key hash of ad=].
+  : <dfn>ad component render url hashes</dfn>
+  :: A [=list=] of [=SHA-256=] hashes for each of the ad components in the
+     winning bid, as calculated using [=compute the key hash of component ad=].
+  : <dfn>reporting id hash</dfn>
+  :: A [=SHA-256=] hash of the [=k-anonymity key=] for the reporting ID in the
+     winning bid, as calculated using [=compute the key hash of reporting ID=].
+</dl>
+
+<dfn>server auction ghost winner</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="server auction ghost winner">
+  : <dfn>candidates</dfn>
+  :: [=server auction join candidates=] associated with this winner.
+  : <dfn>interest group owner</dfn>
+  :: An [=origin=]. The non-k-anonymous winning bid's interest group [=interest group/owner=].
+  : <dfn>interest group name</dfn>
+  :: A [=string=]. The non-k-anonymous winning bid's interest group [=interest group/name=].
+  : <dfn>ghost winner bid info</dfn>
+  :: Null or a [=server auction ghost winner bid info=]. Contains information needed for
+     ghost winners in component auctions.
+</dl>
+
+A <dfn>server auction ghost winner bid info</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="server auction ghost winner bid info">
+  : <dfn>ad render url</dfn>
+  :: [=URL=]. The [=leading bid info/leading non-k-anon-enforced bid=]'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=] from the
+     auction.
+  : <dfn>ad components</dfn>
+  :: A [=list=] of [=URLs=]. A list of the non-k-anonymous winning bid's

```

Comment:
is "non-k-anonymous winning bid" the same as [=leading bid info/leading non-k-anon-enforced bid=]? If so, probably good to link to it as well, as what you did for render url above.